hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -477,14 +479,19 @@ private Pair<SchemaProvider, Pair<String, JavaRDD<HoodieRecord>>> fetchFromSourc
     }
 
     boolean shouldCombine = cfg.filterDupes || cfg.operation.equals(WriteOperationType.UPSERT);
+    boolean isDropPartitionColumns = props.getBoolean(DROP_PARTITION_COLUMNS.key());
+    String partitionColumns = HoodieSparkUtils.getPartitionColumns(keyGenerator, props);
+    List<String> listOfPartitionColumns = Arrays.asList(partitionColumns.split(","));
     JavaRDD<GenericRecord> avroRDD = avroRDDOptional.get();
+    HoodieKey key = keyGenerator.getKey(avroRDD.first());
     JavaRDD<HoodieRecord> records = avroRDD.map(gr -> {
+      gr = isDropPartitionColumns ? HoodieAvroUtils.removeFields(gr, listOfPartitionColumns) : gr;
```

Comment:
Would be good if we can also also add a UT in `TestHoodieDeltaStreamer`.

Reply:
There are no tests for the whole `readFromSource` or `fetchFromSource`, it will take more time to add UT for the entire method, can I take it up as a separate task, since this PR is a blocker for this release.

Reply:
Filed HUDI-3863 as followup. Feel free to pick it up.